### PR TITLE
Remove remainder of the v1 discovery service

### PR DESCRIPTION
**Pull request by ijsnellf**
Removes unused logic, marks flags as deprecated.

https://github.com/istio/istio/issues/6049

### CODE REVIEW

File: pilot/cmd/pilot-discovery/main.go [hunk 1]
```diff
@@ -150,6 +142,17 @@ func init() {
...
 	discoveryCmd.PersistentFlags().MarkDeprecated("monitoringPort", "Use --monitoringAddr instead")
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.DiscoveryOptions.EnableProfiling, "profile", true,
+		"Enable profiling via web interface host:port/debug/pprof")
+	discoveryCmd.PersistentFlags().MarkDeprecated("profile", "")
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.DiscoveryOptions.EnableCaching, "discovery_cache", true,
+		"Enable caching discovery service responses")
+	discoveryCmd.PersistentFlags().MarkDeprecated("discovery_cache", "")
+	// TODO (rshriram): Need v1/v2 endpoints and option to selectively
+	// enable webhook for specific xDS config (cds/lds/etc).
+	discoveryCmd.PersistentFlags().StringVar(&serverArgs.DiscoveryOptions.WebhookEndpoint, "webhookEndpoint", "",
```
Comment (ijsnellf) [inline pilot/cmd/pilot-discovery/main.go]: @rshriram: please confirm that this a v1 only flag.

Comment (ijsnellf) [reply pilot/cmd/pilot-discovery/main.go]: it isn't used anywhere.

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -28,6 +28,7 @@ import (
 
 	"code.cloudfoundry.org/copilot"
 	"github.com/davecgh/go-spew/spew"
+	"github.com/emicklei/go-restful"
```
Comment (zachgersh) [inline pilot/pkg/bootstrap/server.go]: did you check and see if this dependency is used anywhere else? May be able to remove it from vendor?

File: pilot/cmd/pilot-discovery/main.go (hunk 1 above)
Comment (rshriram) [reply pilot/cmd/pilot-discovery/main.go]: Its v1 only.. and I sugest removing this..

File: pilot/cmd/pilot-discovery/main.go [hunk 3]
```diff
@@ -150,6 +142,17 @@ func init() {
 	discoveryCmd.PersistentFlags().IntVar(&monitoringPort, "monitoringPort", 9093,
 		"HTTP port to use for the exposing pilot self-monitoring information")
 	discoveryCmd.PersistentFlags().MarkDeprecated("monitoringPort", "Use --monitoringAddr instead")
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.DiscoveryOptions.EnableProfiling, "profile", true,
+		"Enable profiling via web interface host:port/debug/pprof")
+	discoveryCmd.PersistentFlags().MarkDeprecated("profile", "")
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.DiscoveryOptions.EnableCaching, "discovery_cache", true,
+		"Enable caching discovery service responses")
+	discoveryCmd.PersistentFlags().MarkDeprecated("discovery_cache", "")
```
Comment (rshriram) [inline pilot/cmd/pilot-discovery/main.go]: remove

File: pilot/cmd/pilot-discovery/main.go [hunk 4]
```diff
@@ -150,6 +142,17 @@ func init() {
 	discoveryCmd.PersistentFlags().IntVar(&monitoringPort, "monitoringPort", 9093,
 		"HTTP port to use for the exposing pilot self-monitoring information")
 	discoveryCmd.PersistentFlags().MarkDeprecated("monitoringPort", "Use --monitoringAddr instead")
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.DiscoveryOptions.EnableProfiling, "profile", true,
+		"Enable profiling via web interface host:port/debug/pprof")
```
Comment (rshriram) [inline pilot/cmd/pilot-discovery/main.go]: remove?

File: pilot/cmd/pilot-discovery/main.go [hunk 5]
```diff
@@ -150,6 +142,17 @@ func init() {
...
+	discoveryCmd.PersistentFlags().MarkDeprecated("profile", "")
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.DiscoveryOptions.EnableCaching, "discovery_cache", true,
+		"Enable caching discovery service responses")
+	discoveryCmd.PersistentFlags().MarkDeprecated("discovery_cache", "")
+	// TODO (rshriram): Need v1/v2 endpoints and option to selectively
+	// enable webhook for specific xDS config (cds/lds/etc).
+	discoveryCmd.PersistentFlags().StringVar(&serverArgs.DiscoveryOptions.WebhookEndpoint, "webhookEndpoint", "",
+		"Webhook API endpoint (supports http://sockethost, and unix:///absolute/path/to/socket")
+	discoveryCmd.PersistentFlags().MarkDeprecated("webhookEndpoint", "")
```
Comment (rshriram) [inline pilot/cmd/pilot-discovery/main.go]: remove. No deprecation

File: pilot/pkg/bootstrap/server.go [hunk 6]
```diff
@@ -156,9 +157,36 @@ type ServiceArgs struct {
...
+	// The listening address for the monitoring port. If the port in the address is empty or "0" (as in "127.0.0.1:" or "[::1]:0")
+	// a port number is automatically chosen.
+	MonitoringAddr string
+
+	// Deprecated
+	EnableProfiling bool
+	// Deprecated
+	EnableCaching bool
+	// Deprecated
+	WebhookEndpoint string
```
Comment (rshriram) [inline pilot/pkg/bootstrap/server.go]: remove all deprecated

**Comment by ijsnellf** [comment]
@rshriram: removed deprecated flags as requested.

File: install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml [hunk 7]
```diff
@@ -38,7 +38,7 @@ spec:
           - containerPort: 15010
           readinessProbe:
             httpGet:
-              path: /v1/registration
+              path: /debug/adsz
```
Comment (ijsnellf) [inline install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml]: I switched this to adsz. We don't have a proper `/health` API.

**Comment by ijsnellf** [comment]
Waiting on https://github.com/istio/istio/pull/6936

File: install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml [hunk 8]
```diff
@@ -42,7 +42,7 @@ spec:
           - containerPort: 15010
           readinessProbe:
             httpGet:
-              path: /v1/registration
+              path: /debug/edsz
```
Comment (ijsnellf) [inline install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml]: roughly equivalent new api

Comment (ijsnellf) [reply install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml]: we probably want to use something else in the future, if we have 10k endpoints, we don't necessarily want to marshal them all to json for no reason.

File: pilot/pkg/bootstrap/server.go [hunk 9]
```diff
@@ -732,6 +752,31 @@ func (s *Server) initConfigRegistry(serviceControllers *aggregate.Controller) {
 	})
 }
 
+// Flush cached discovery responses whenever services, service instances, or routing configuration changes.
+// TODO: smarter cache invalidation. this is very simplistic.
+// essentially we invalidate our cache whenever ANYTHING changes.
+func (s *Server) registerClearCacheHandlers() error {
```
Comment (ijsnellf) [inline pilot/pkg/bootstrap/server.go]: moved this logic from inside `NewDiscoveryService(...)`

File: pilot/cmd/pilot-discovery/main.go [hunk 10]
```diff
@@ -130,14 +130,6 @@ func init() {
 		"Discovery service grpc address, with https")
 	discoveryCmd.PersistentFlags().StringVar(&serverArgs.DiscoveryOptions.MonitoringAddr, "monitoringAddr", ":9093",
 		"HTTP address to use for the exposing pilot self-monitoring information")
-	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.DiscoveryOptions.EnableProfiling, "profile", true,
```
Comment (costinm) [inline pilot/cmd/pilot-discovery/main.go]: What happened with this flag ? We should still be able to turn off pprof and debug interfaces.

File: pilot/pkg/bootstrap/server.go (hunk 2 above)
Comment (costinm) [inline pilot/pkg/bootstrap/server.go]: Oh no - can you just use regular handlers ? We no longer have 'restful' needs.

File: pilot/pkg/bootstrap/server.go [hunk 11]
```diff
@@ -126,9 +127,29 @@ type ServiceArgs struct {
 	Consul     ConsulArgs
 }
 
+// DiscoveryServiceOptions contains options for create a new discovery
+// service instance.
+type DiscoveryServiceOptions struct {
+	// The listening address for HTTP. If the port in the address is empty or "0" (as in "127.0.0.1:" or "[::1]:0")
+	// a port number is automatically chosen.
+	HTTPAddr string
```
[the remark this answers is not in the document]
Comment (rshriram) [reply pilot/pkg/bootstrap/server.go]: Yes.. lets get rid of all of these unused ports

**Comment by rshriram** [comment]
Please get rid of webhooks

File: pilot/pkg/bootstrap/server.go (hunk 2 above)
Comment (ijsnellf) [reply pilot/pkg/bootstrap/server.go]: I'll get rid of it once I get tests passing. I seem to have broken a bunch of things, and not sure why.

File: pilot/pkg/bootstrap/server.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (ijsnellf) [reply pilot/pkg/bootstrap/server.go]: Can do.

File: pilot/cmd/pilot-discovery/main.go (hunk 10 above)
Comment (ijsnellf) [reply pilot/cmd/pilot-discovery/main.go]: It was only used in v1 discovery. But if we want to keep the flag around for future use, that makes sense.

**Comment by ijsnellf** [comment]
@rshriram: Yep, this should nuke webhooks (at least in v1 discovery).

**Comment by costinm** [comment]
What's the status ? We can fix some of the comments in the review in a separate PR, but tests are not passing. Maybe move this to master if it's not ready by Fri ?

**Comment by costinm** [comment]
Moved to master, not P0 for 1.0

**Comment by ijsnellf** [comment]
This has gotten quite out of date. I'm closing it in favor of a fresh PR.

**Pull request closed without merging**